std/crypto/tls/conn.go

```diff
@@ -1574,10 +1591,14 @@ func (c *Conn) release() {
 
 // Close closes the connection.
 func (c *Conn) Close() error {
-	c.closeMux.Lock()
+	if !c.hasLock {
```

Comment:
here is still not a guard for concurrent operations:
1. another goroutine is locking, so we get a `c.hasLock == true` here and not call c.closeMux.Lock()